migration/transitioning.rst

```diff
@@ -384,9 +391,233 @@ currently logged in customer - even though this is complex analytical
+will present a problem in any distributed system, since it’s difficult
+to make sure that no two nodes accept the same unique value. Enforcing
+the constraint would require expensive scans of the data across all
+nodes.
+
+To solve this problem, for the models which are logically related
+to a store (the tenant for our app), you should add store\_id to
+the constraints, effectively scoping objects unique inside a given
+store. This helps add the concept of tenancy to your models, thereby
+making the multi-tenant system more robust.
+
+Let's begin by adjusting our model definitions and have Django generate a new migration for the two changes discussed.
+
+.. code-block:: python
+
+  from django.utils import timezone
+  from django.db import models
+
+  class Store(models.Model):
+    name = models.CharField(max_length=255)
+    url = models.URLField()
+
+  class Product(models.Model):
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    price = models.DecimalField(max_digits=6, decimal_places=2),
+    quantity = models.IntegerField()
+    store = models.ForeignKey(Store)
+
+    class Meta(object):                  # added
+      unique_together = ["id", "store"]  #
+
+  class Purchase(models.Model):
+    ordered_at = models.DateTimeField(default=timezone.now)
+    billing_address = models.TextField()
+    shipping_address = models.TextField()
+
+    product = models.ForeignKey(
+      Product,
+      db_constraint=False                # added
+    )
+    store = models.ForeignKey(Store)     # added
+
+    class Meta(object):                  # added
+      unique_together = ["id", "store"]  #
+
+Create a migration to reflect the change: :code:`./manage.py makemigrations`.
+
+Next we need some custom migrations to adapt the existing key structure in the database for compatibility with Citus. To keep these migrations separate from the ones for the ordinary application, we'll make a new citus application in the same Django project.
+
+.. code-block:: bash
+
+  # Make a new sub-application in the project
+  django-admin startapp citus
+
+Edit :code:`appname/settings.py` and add :code:`'citus'` to the array :code:`INSTALLED_APPS`.
+
+Next we'll add a custom migration to remove simple primary keys which will become composite: :code:`./manage.py makemigrations citus --empty --name remove_simple_pk`. Edit the result to look like this:
+
+.. code-block:: python
+
+  from __future__ import unicode_literals
+  from django.db import migrations
+
+  class Migration(migrations.Migration):
+    dependencies = [
+      ('appname', '<name of latest migration>')
+    ]
+
+    operations = [
+      # Django considers "id" the primary key of these tables, but
+      # the database mustn't, because the primary key will be composite
+      migrations.RunSQL(
+        "ALTER TABLE mtdjango_product DROP CONSTRAINT mtdjango_product_pkey;",
+        "ALTER TABLE mtdjango_product ADD CONSTRAINT mtdjango_product_pkey PRIMARY KEY (id)"
```

Comment:
I think it should be:

"ALTER TABLE mtdjango_product ADD CONSTRAINT mtdjango_product_pkey PRIMARY KEY (store_id,id)"
(OR)
It could be removed(considering we have NOT NULL constraint for both store_id,id) because we already have unique_together = ["id", "store"]

Reply:
D'oh, good catch.